app/code/Magento/Catalog/Model/CategoryRepository.php

```diff
@@ -82,6 +82,10 @@ public function save(\Magento\Catalog\Api\Data\CategoryInterface $category)
         $existingData = array_diff_key($existingData, array_flip(['path', 'level', 'parent_id']));
         $existingData['store_id'] = $storeId;
 
+        if (is_array($category->getData())) {
+            $existingData = array_replace($existingData, $category->getData());
```

Comment:
At first glance I don't see a problem with such approach, we just need to check the failed tests to see the impact of such change.